lib/customer.rb

```diff
@@ -0,0 +1,43 @@
+require 'csv'
+require 'pry'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all(file_name = 'data/customers.csv')
+    customers = []
+    clients = CSV.read(file_name, headers: true).map(&:to_h)
```

Comment:
It's worth thinking about how you would solve this problem if you couldn't edit the csv to have headers.